scripts/pre_commit_linter.py

```diff
@@ -468,6 +468,57 @@ def _pre_commit_linter(all_files):
     return summary_messages
 
 
+def _check_newline_character(all_files):
+    """This function is used to check new line characters.
+    """
+    print 'Starting to check new line at EOF'
+    print '----------------------------------------'
+    total_files_checked = 0
+    total_error_count = 0
+    summary_messages = []
+    all_files = [
+        filename for filename in all_files if not
+        any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)]
+    failed = False
+    for filename in all_files:
+        with open(filename, 'rb+') as f:
+            total_files_checked += 1
+            chars = 0
+            for line in f:
+                chars += len(line)
+            if chars == 1:
+                failed = True
+                print '%s --> Error:Only one character in file' % filename
+                total_error_count += 1
+            elif chars > 1:
+                f.seek(-2, 2)
+                if not(f.read(1) != '\n' and f.read(1) == '\n'):
+                    failed = True
+                    print '%s --> Please add/remove extra newline' % filename
```

Comment:
This isn't clear. "Please ensure that this file ends with exactly one newline char."